net_util/src/tap.rs

```diff
@@ -249,9 +249,26 @@ impl Tap {
 
         let mut ifreq = self.get_ifreq();
 
+        #[allow(clippy::cast_lossless)]
+        let ret =
+            unsafe { ioctl_with_ref(&sock, net_gen::sockios::SIOCGIFFLAGS as c_ulong, &ifreq) };
+        if ret < 0 {
+            return Err(Error::IoctlError(IoError::last_os_error()));
+        }
+
+        // If TAP device is already up don't try and enable it
+        let ifru_flags = unsafe { ifreq.ifr_ifru.ifru_flags.as_ref() };
+        if ifru_flags
+            & (net_gen::net_device_flags_IFF_UP | net_gen::net_device_flags_IFF_RUNNING) as i16
```

Comment:
Are you trying to check that if both of these flags are set, then you can return? If that's the case the check is wrong as any of these condition being set will consider the TAP interface up and ready.
And again if that's the case, then I suggest using:
```rust
if ifru_flags
        & (net_gen::net_device_flags_IFF_UP | net_gen::net_device_flags_IFF_RUNNING) as i16
        == (net_gen::net_device_flags_IFF_UP | net_gen::net_device_flags_IFF_RUNNING) {
```

Reply:
Urgh, yes. Will fix.